back-end/src/test/java/com/codurance/guru/AcceptanceAPIShould.java

```diff
@@ -0,0 +1,34 @@
+package com.codurance.guru;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = GuruApplication.class)
+@TestPropertySource(value={"classpath:application.properties"})
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+public class AcceptanceAPIShould {
+
+    TestRestTemplate restTemplate = new TestRestTemplate();
+    HttpHeaders headers = new HttpHeaders();
+
+    @Test
+    public void retrieve_a_craftsperson() throws Exception{
+        HttpEntity<String> entity = new HttpEntity<String>(null, headers);
+        ResponseEntity<String> response = restTemplate.exchange(
+                "http://localhost:8080/craftspeople/1", HttpMethod.GET,entity, String.class);
+        String expected = "{\"id\":1,\"firstName\":\"Jose\",\"lastName\":\"Campos\",\"imageUrl\":null}";
```

Comment:
I can't remember if it was hard but I guess that working with an object and serialising it to JSON reduces the visual clutter!